src/main/java/rx/internal/operators/OperatorMerge.java

```diff
@@ -246,63 +246,58 @@ private void handleScalarSynchronousObservable(ScalarSynchronousObservable<? ext
 
         private void handleScalarSynchronousObservableWithoutRequestLimits(ScalarSynchronousObservable<? extends T> t) {
             T value = t.get();
-            if (getEmitLock()) {
-                boolean moreToDrain;
-                try {
-                    actual.onNext(value);
-                } finally {
-                    moreToDrain = releaseEmitLock();
-                }
-                if (moreToDrain) {
-                    drainQueuesIfNeeded();
-                }
-                request(1);
-                return;
-            } else {
-                try {
-                    getOrCreateScalarValueQueue().onNext(value);
-                } catch (MissingBackpressureException e) {
-                    onError(e);
+            try {
+                synchronized (this) {
+                    if (emitLock) {
+                        missedEmitting = true;
+                        getOrCreateScalarValueQueue().onNext(value);
+                        return;
+                    }
+                    missedEmitting = false;
+                    emitLock = true;
                 }
+            } catch (MissingBackpressureException e) {
+                onError(e);
                 return;
             }
+            boolean moreToDrain;
+            try {
+                actual.onNext(value);
+            } finally {
+                moreToDrain = releaseEmitLock();
+            }
+            if (moreToDrain) {
+                drainQueuesIfNeeded();
+            }
+            request(1);
```

Comment:
In a future rewrite we can optimize this because it could drain queues and request(n) and then immediately request(1) again.